trunk/src/app/srs_app_http_stream.cpp

```diff
@@ -630,6 +630,8 @@ srs_error_t SrsLiveStream::serve_http(ISrsHttpResponseWriter* w, ISrsHttpMessage
     // Correct the app and stream by path, which is created from template.
     // @remark Be careful that the stream has extension now, might cause identify fail.
     req->stream = srs_path_basename(r->path());
+    // remove the extension of stream if have. for instance, test.flv -> test
+    req->stream = srs_path_filename(req->stream);
```

Comment:
If remove the extension, the stat API can't know which type of stream, RTMP or FLV or TS.